micro_http/src/common/headers.rs

```diff
@@ -13,39 +17,178 @@ pub enum Header {
     ContentLength,
     /// Header `Content-Type`.
     ContentType,
+    /// Header `Expect`.
+    Expect,
+    /// Header `Transfer-Encoding`.
+    TransferEncoding,
 }
 
 impl Header {
     fn raw(&self) -> &'static [u8] {
         match self {
             Header::ContentLength => b"Content-Length",
             Header::ContentType => b"Content-Type",
+            Header::Expect => b"Expect",
+            Header::TransferEncoding => b"Transfer-Encoding",
+        }
+    }
+
+    fn try_from(string: &[u8]) -> Result<Self, RequestError> {
+        if let Ok(s) = String::from_utf8(string.to_vec()) {
+            match s.trim() {
+                "Content-Length" => Ok(Header::ContentLength),
+                "Content-Type" => Ok(Header::ContentType),
+                "Expect" => Ok(Header::Expect),
+                "Transfer-Encoding" => Ok(Header::TransferEncoding),
+                _ => Err(RequestError::InvalidHeader),
+            }
+        } else {
+            Err(RequestError::InvalidHeader)
         }
     }
 }
 
 /// Wrapper over the list of headers associated with a Request/Response.
 #[derive(Debug)]
 pub struct Headers {
-    headers: HashMap<Header, String>,
+    optional_headers: HashMap<Header, String>,
+    content_length: i32,
+    expect: bool,
+    chunked: bool,
 }
 
 impl Headers {
     /// By default Requests are created with no headers.
     pub fn default() -> Headers {
         Headers {
-            headers: HashMap::new(),
+            optional_headers: HashMap::new(),
+            content_length: 0,
+            expect: false,
+            chunked: false,
         }
     }
 
     /// Adds a new header to the list.
     pub fn add(&mut self, header: Header, value: String) {
-        self.headers.insert(header, value);
+        self.optional_headers.insert(header, value);
+    }
+
+    /// Expects one header line and parses it, updating the header structure or returning an
+    /// error if the header is invalid.
+    pub fn parse_header_line(&mut self, header_line: &[u8]) -> Result<bool, RequestError> {
```

Comment:
It's weird for `parse_header_line` to return a Result of bool. If you update the code as per the suggestion in `src/request.rs`, this will not be needed. Let's discuss about this offline.